og_ui/src/Controller/SubscriptionController.php

```diff
@@ -0,0 +1,163 @@
+    $entity_access = $this->entityTypeManager()->getAccessControlHandler($entity_type_id);
+    $group = $entity_storage->load($entity_id);
+
+    $account = User::load($this->currentUser()->id());
+    $field_name = $request->query->get('field_name');
+
+    if (empty($field_name)) {
+      $field_name = og_get_best_group_audience_field('user', $account, $entity_type, $bundle);
+      if (empty($field_name)) {
+        throw new NotFoundHttpException();
+      }
+    }
+
+    // @todo Requires FieldableEntityInterface/ContentEntityInterface
+    $field = $group->getFieldDefinition($field_name);
+
+    if (empty($instance) || !$entity_access->fieldAccess('view', $field, $account)) {
+      // Field name given is incorrect, or user doesn't have access to the field.
+      throw new NotFoundHttpException();
+    }
+
+    if ($account->isAnonymous()) {
+      // Anonymous user can't request membership.
+      $destination = $this->getDestinationArray();
+
+      $user_login_url = Url::fromRoute('user.login', [], $destination);
+
+      // @todo I think this is correct? Other options are visitors or require
+      // approval both of which apply to the else instead of here.
+      if ($this->config('user.settings')->get('register') === USER_REGISTER_ADMINISTRATORS_ONLY) {
+        drupal_set_message($this->t('In order to join any group, you must <a href=":login">login</a>. After you have successfully done so, you will need to request membership again.', [':login' => $user_login_url]));
+      }
+      else {
+        $user_register_url = Url::fromRoute('user.register', [], $destination);
+        drupal_set_message($this->t('In order to join any group, you must <a href=":login">login</a> or <a href=":register">register</a> a new account. After you have successfully done so, you will need to request membership again.', [':register' => $user_register_url, ':login' => $user_login_url]));
+      }
+
+      return new RedirectResponse(Url::fromRoute('user.page')->setAbsolute(TRUE)->toString());
+    }
+
+    $redirect = FALSE;
+    $message = '';
+    $params = [
+      '@user' => $account->getDisplayName(),
+    ];
+
+    // Show the group name only if user has access to it.
+    $params['@group'] = $group->access('view', $account) ?  $group->label() : $this->t('Private group');
+
+    if (Og::isMember($group, $account, [OG_STATE_BLOCKED])) {
+      // User is blocked, access denied.
+      throw new AccessDeniedHttpException();
+    }
+
+    if (Og::isMember($group, $account, [OG_STATE_PENDING])) {
+      // User is pending, return them back.
+      // @todo Amitai, what is the purpose of this $user->uid == $user->uid
+      // check? This will always be true?!
+      $message = $account->id() == $account->id() ? $this->t('You already have a pending membership for the group @group.', $params) : $this->t('@user already has a pending membership for the  the group @group.', $params);
```

Comment:
No reason, seems like a (harmless) bug that sneaked in from a wrong copy/ paste :)